auth/auth-core/src/main/java/com/bazaarvoice/emodb/auth/apikey/ApiKey.java

```diff
@@ -13,12 +14,29 @@
  */
 public class ApiKey extends AuthIdentity {
 
-    public ApiKey(String key, Set<String> roles) {
-        super(key, roles);
+    public ApiKey(String key, String internalId, Set<String> roles) {
+        super(key, internalId, roles);
     }
 
     @JsonCreator
-    public ApiKey(@JsonProperty("id") String key, @JsonProperty("roles") List<String> roles) {
-        this(key, ImmutableSet.copyOf(roles));
+    public ApiKey(@JsonProperty("id") String key,
+                  @JsonProperty("internalId") String internalId,
```

Comment:
It's not nullable but it has to accept nulls to grandfather in existing records.